src/transformers/models/colpali/modeling_colpali.py

```diff
@@ -0,0 +1,655 @@
+        return model_inputs
+
+
+@dataclass
+class ColPaliOutput(ModelOutput):
+    """
+    Base class for ColPali embeddings output.
+
+    Args:
+        embeddings (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`):
+            The embeddings of the model.
+    """
+
+
+COLPALI_START_DOCSTRING = r"""
+    ColPali is a PaliGemma variant to produce multi-vector representations from images.
+    It was introduced in the paper [ColPali: Efficient Document Retrieval with Vision Language Models](https://arxiv.org/abs/2407.01449).
+
+    ### Resources
+    - A blog post detailing ColPali, a vision retrieval model, can be found [here](https://huggingface.co/blog/manu/colpali). 🌎
+    - The training codebase for ColPali can be found [here](https://github.com/illuin-tech/colpali). 🌎
+"""
+
+COLPALI_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`):
+            Indices of input sequence tokens in the vocabulary. Padding will be ignored by default should you provide
+            it.
+            Indices can be obtained using [`AutoTokenizer`]. See [`PreTrainedTokenizer.encode`] and
+            [`PreTrainedTokenizer.__call__`] for details.
+            [What are input IDs?](../glossary#input-ids)
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, image_size, image_size)):
+            The tensors corresponding to the input images. Pixel values can be obtained using
+            [`AutoImageProcessor`]. See [`SiglipImageProcessor.__call__`] for details ([]`PaliGemmaProcessor`] uses
+            [`SiglipImageProcessor`] for processing images). If none, ColPali will only process text (query embeddings).
+        attention_mask (`torch.Tensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+            - 1 for tokens that are **not masked**,
+            - 0 for tokens that are **masked**.
+            [What are attention masks?](../glossary#attention-mask)
+            Indices can be obtained using [`AutoTokenizer`]. See [`PreTrainedTokenizer.encode`] and
+            [`PreTrainedTokenizer.__call__`] for details.
+            If `past_key_values` is used, optionally only the last `decoder_input_ids` have to be input (see
+            `past_key_values`).
+            If you want to change padding behavior, you should read [`modeling_opt._prepare_decoder_attention_mask`]
+            and modify to your needs. See diagram 1 in [the paper](https://arxiv.org/abs/1910.13461) for more
+            information on the default strategy.
+            - 1 indicates the head is **not masked**,
+            - 0 indicates the head is **masked**.
+"""
+
+
+@add_start_docstrings(
+    COLPALI_START_DOCSTRING,
+    "Adapter from colpali-engine==0.3.0: https://github.com/illuin-tech/colpali.",
+)
+class ColPaliModel(PaliGemmaPreTrainedModel):
```

Comment:
[ModelName]Model is usually reserved to "encoder" or model which outputs non-task specific embeddings, is that the case here? Otherwise we could rename to something like ColPaliForRetrieval